streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -685,6 +691,7 @@ private void recycleTask(final Task task,
                              final Map<TaskId, RuntimeException> taskExceptions) {
         Task newTask = null;
         try {
+            task.suspend();
```

Comment:
Is it assuming that suspend is idempotent here? Without state updater would we call it twice, in `handleRevocation` and `handleAssignment`?

Reply:
Yes, I agree. That was actually the starting point of our discussion the other day. I will remove it.

Reply:
`recycleTask()` is only called from the code path that uses the state updater and in that code path we do not suspend active tasks on revocation. The only place we suspend tasks (active and standbys) coming from the state updater is here. In the code path without state updater, standbys are suspended before recycling but that code path is not executed with state updater enabled. During shutdown and error handling tasks are suspended in the code path that is executed with and without state updater.
So I think the answer to your questions is "no" and "no".

Reply:
Ah right! Thanks.

In that case could we change the func name to `recycleTaskFromStateUpdater` to be clearer?

Reply:
I will rename the method in my next PR.